src/hooks/useEarnings.js

```diff
@@ -34,5 +35,13 @@ export default function useEarnings() {
     setEarnings([...earnings.slice(0, earnings.length - 1), newEarning])
   }
 
-  return { lastEarning, distribute, addEarning, earnings }
+  function spendCoins(type, value) {
+    const lastSpending = {
+      ...lastEarning,
+      [type]: lastEarning[type] + value,
+    }
+    setEarnings([...earnings.slice(0, earnings.length - 1), lastSpending])
```

Comment:
Hm.. a spending is not an earning & allocation of coins. If you want to model spending and earning & allocation as a list of transactions maybe rename this.